### PR TITLE
Added junit tests for login usecase interactor. 

### PR DESCRIPTION
**Pull request by vrindasubhash**
Added a few test cases and confirmed the login use case interactor works as expected.

Tests:
1) Existing user logins in with correct password
2) Non-existent user tries to login
3) Wrong password for existing user

### CODE REVIEW

**Comment by merrickliu888** [comment]
@vrindasubhash could you attach a screen shot of your coverage with the tests? Thanks!

**Comment by vrindasubhash** [comment]
Screenshot 2023-11-19 at 2 41 17 AM

File: test/use_case/login/LoginInteractorTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+public class LoginInteractorTest {
+    MemoryUserDataAccessObject userDataAccessObject;
+    String firstUser = "Bob";
+    String firstPassword = "pass";
+
+    @Before
+    public void setUp() throws Exception {
+        this.userDataAccessObject = new MemoryUserDataAccessObject();
+        this.userDataAccessObject.saveUser(new CommonUser(this.firstUser, this.firstPassword, null));
```
Comment (merrickliu888) [inline test/use_case/login/LoginInteractorTest.java:19]: Don't use the `CommonUser` constructor, use the `UserFactory` which can also create a default user with default preferences.

File: test/use_case/login/LoginInteractorTest.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        this.userDataAccessObject = null;
+    }
+
+    @Test
+    public void testUserExistsAndPasswordCorrect() {
```
Comment (merrickliu888) [inline test/use_case/login/LoginInteractorTest.java:28]: Test the `userPreferences` values too.

File: test/use_case/login/LoginInteractorTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    String firstPassword = "pass";
+
+    @Before
+    public void setUp() throws Exception {
+        this.userDataAccessObject = new MemoryUserDataAccessObject();
+        this.userDataAccessObject.saveUser(new CommonUser(this.firstUser, this.firstPassword, null));
+    }
+
+    @After
+    public void tearDown() throws Exception {
```
Comment (merrickliu888) [inline test/use_case/login/LoginInteractorTest.java:23]: Not necessary in this case.

File: test/use_case/login/LoginInteractorTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import static org.junit.Assert.*;
+
+public class LoginInteractorTest {
+    MemoryUserDataAccessObject userDataAccessObject;
+    String firstUser = "Bob";
+    String firstPassword = "pass";
+
+    @Before
+    public void setUp() throws Exception {
+        this.userDataAccessObject = new MemoryUserDataAccessObject();
```
Comment (merrickliu888) [inline test/use_case/login/LoginInteractorTest.java:18]: You can instantiate this above when you first declare this `userDataAccessObject`, so you don't have to reinstantiate it each time.

**Review by merrickliu888 — CHANGES_REQUESTED** [review]
Overall looks good. Pretty much got everything, just some small changes.

**Review by merrickliu888 — APPROVED** [review]
Good changes. Thanks for removing the import.

**Comment by vrindasubhash** [comment]
Screenshot 2023-11-20 at 2 03 16 PM

Coverage now 100%

**Pull request merged**